test/common/stats/stats_impl_test.cc

```diff
@@ -466,5 +468,35 @@ TEST(TagProducerTest, CheckConstructor) {
       "No regex specified for tag specifier and no default regex for name: 'test_extractor'");
 }
 
+TEST(SomeTest, HotAllocConsistencyTest) {
+  int max_stats_ = 16384; // ENVOY_DEFAULT_MAX_STATS
+
+  SharedMemoryHashSetOptions options_ = Server::sharedMemHashOptions(max_stats_);
+  uint8_t* ptr = static_cast<uint8_t*>(
+      calloc(SharedMemoryHashSet<Stats::RawStatData>::numBytes(options_), max_stats_));
+  std::unique_ptr<RawStatDataSet> stats_set_ =
```

Comment:
Since you have your fancy new block allocator, why not use it here? :) .  Otherwise the test is really testing a reimplementation of your allocator rather than the allocator itself.

Reply:
Good catch. Rewrote to look like
```
TEST(SomeTest, HotAllocConsistencyTest) { 
  int max_stats_ = 16384; // ENVOY_DEFAULT_MAX_STATS

  SharedMemoryHashSetOptions options_ = Server::sharedMemHashOptions(max_stats_);
  uint8_t* ptr = static_cast<uint8_t*>(
      calloc(SharedMemoryHashSet<Stats::RawStatData>::numBytes(options_), max_stats_));

  Stats::BlockRawStatDataAllocator allocator = 
    Stats::BlockRawStatDataAllocator(options_, true, ptr);                                                                                                                                                                      

  auto data_1 = allocator.alloc("foo");
  auto data_2 = allocator.alloc("foo");

  EXPECT_EQ(data_1, data_2);

  ::free(ptr);
}
```